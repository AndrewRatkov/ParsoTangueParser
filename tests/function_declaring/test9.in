def int id(int x)
if x + 30 == 0
then return 30+30;
fi

if id(x - 1)
then return id(x - 2);
fi

return 0;

enddef
