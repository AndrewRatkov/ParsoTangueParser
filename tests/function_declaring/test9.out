[DEF FUN id(INTEGER x)=>INT]
|   
+-->IF
|   |
|   +--->[INT,==]------->[INT,+ ]------->[INT,x]
|   |        |               |
|   |        |               +---------->[INT,30]
|   |        |
|   |        +---------->[INT,0]
|   |
|   THEN
|   |
|   +--->[RETURN]--->[INT,+ ]------->[INT,30]
|   |                    |
|   |                    +---------->[INT,30]
|   |
|   FI
|   
+-->IF
|   |
|   +--->[CALL INT id]------->[INT,- ]------->[INT,x]
|   |                             |
|   |                             +---------->[INT,1]
|   |
|   THEN
|   |
|   +--->[RETURN]--->[CALL INT id]------->[INT,- ]------->[INT,x]
|   |                                         |
|   |                                         +---------->[INT,2]
|   |
|   FI
|   
+-->[RETURN]--->[INT,0]
